### ISSUE TITLE
Add Object>>perform:with:

### ISSUE THREAD

**Issue by hhzl** [amber-smalltalk/amber#1180]
Pharo has 

        Object>>perform:with:        

Amber only has

     #perform:withArguments

The implementation is

    perform: aSymbol with: anObject 
	        "Send the selector, aSymbol, to the receiver with anObject as its argument.
	
           	^ self perform: aSymbol withArguments: (Array with: anObject)

**Comment by herby** [comment]
I would see `{anObject}` instead `Array with: anObject`.

**Comment by herby** [comment]
BTW does it only have `perform:with:`? Doesn't it have `perform:with:with:` etc. If yes, I would see it as all or nothing, getting only part would be confusing.

**Comment by hhzl** [comment]
![pharo4 0-object_perform_with_screenshot_2015-05-25_15 15](https://cloud.githubusercontent.com/assets/1112023/7799072/2f5e2f4a-02f1-11e5-9c82-47f640bff1ac.png)

And interestingly an implementation with array notation

![pharo4 0_perform_with_3_screenshot_2015-05-25_15 16](https://cloud.githubusercontent.com/assets/1112023/7799082/47be26c6-02f1-11e5-806c-a37410571c3d.png)

**Comment by herby** [comment]
Yes, so we should include all three. Just a PR is missing (and I would use `{...}` in one `with:` as well; if for nothing else than for performance).

**Comment by hhzl** [comment]
Amber has

````
basicPerform: aString
	^ self basicPerform: aString withArguments: #()
````

and

````
basicPerform: aString withArguments: aCollection
	<return self[aString].apply(self, aCollection);>
````

in category ``message handling`` of class ``Object``. So Strings are expected instead of Symbols.

**Comment by herby** [comment]
@hhzl What's the point / problem / issue?

  1. You can write aSymbol as well. In Amber, `String == Symbol`, the latter being just an alias name for the former.
  2. `basicPerform` is a very low-level thing that is used to call JS API. It's not actually meant to be used while wearing the "Smalltalk hat". IOW, IMO it does not relate with the topic of #1180.
